File: 1695-maximum-erasure-value/1695-maximum-erasure-value.py
class Solution:
    def maximumUniqueSubarray(self, nums: List[int]) -> int:
        left, seen, curr, ans = 0, set(), 0, 0

        for right in range(len(nums)):
            while nums[right] in seen:
                curr -= nums[left]
                seen.remove(nums[left])
                left += 1
            
            curr += nums[right]
            seen.add(nums[right])
            ans = max(ans, curr)

        return ans

File: 1695-maximum-erasure-value/1695-maximum-erasure-value.cpp
class Solution {
public:
    int maximumUniqueSubarray(vector<int>& nums) {
        unordered_set<int> seen;
        int left = 0, curr = 0, ans = 0;

        for (int right = 0; right < nums.size(); ++right){
            while (seen.count(nums[right])){
                seen.erase(nums[left]);
                curr -= nums[left++];
            }

            seen.insert(nums[right]);
            curr += nums[right];
            ans = max(ans, curr);
        }

        return ans;
    }
};